homeassistant/components/device_sun_light_trigger.py

```diff
@@ -24,21 +25,32 @@
 CONF_DEVICE_GROUP = 'device_group'
 
 
-# pylint: disable=too-many-branches
 def setup(hass, config):
     """ Triggers to turn lights on or off based on device precense. """
+    logger = logging.getLogger(__name__)
 
-    disable_turn_off = 'disable_turn_off' in config[DOMAIN]
+    for key, domain_config in get_configs_for_domain(config, DOMAIN, logger):
```

Comment:
We're moving away from having to specify `switch`, `switch 2`, `switch 3` etc. This was needed because the old `home-assistant.conf` was based on Pythons ConfigParser. We recently made the switch to a YAML-based `configuration.yaml`. YAML supports lists and nested hashes, making the old approach unnecessary. So I would prefer not to have the new method `get_configs_for_domain` added.

Instead, look if `config[DOMAIN]` is a dict (= 1 config given) or a list of dicts (2+ configs given) and handle accordingly.

A config with multiple setups would then look like this:
```YAML
device_sun_light_trigger:
  - light_group: group.living_room
    device_group: device_tracker.Paulus
  - light_group: group.bedroom
    device_group: device_tracker.all_devices
```